src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -1406,26 +1388,25 @@ internal static bool IsProviderQualifiedPath(string path, out string providerId)
                 {
                     // If there is no : then the path is relative to the
                     // current working drive
-
                     result = false;
                     break;
                 }
 
                 // If the :: is the first two character in the path then we
                 // must assume that it is part of the path, and not
                 // delimiting the drive name.
-
                 if (index > 0)
                 {
                     result = true;
 
                     // Get the provider ID
-
-                    providerId = path.Substring(0, index);
-
-                    s_tracer.WriteLine("providerId = {0}", providerId);
+                    providerId = path.Slice(0, index);
+#if DEBUG
```

Comment:
Why are we only writing traces in debug?  This in almost never used.  We need traces for other scenarios other than development.

Reply:
Trace subsystem is very expensive. PerfView shows huge allocations here although _the trace is off_.
Several months ago @daxian-dbw said that he want to review trace code but it is not easy.
I found one sample in our code where we already put a trace call under DEBUG condition.
Until we get new trace code we could follow the sample on hot paths.

Reply:
cc @daxian-dbw Can you please review?

Reply:
@daxian-dbw Friendly ping.

Reply:
@daxian-dbw seems like this still needs a decision, if you have a minute to take a look? 💖